controllers/common/ansible.go

```diff
@@ -62,6 +90,27 @@ func CreateAnsibleJob(policyAutomation *policyv1beta1.PolicyAutomation,
 		mapExtraVars[fieldName] = value
 	}
 
+	// when createnewAnsiblejob is ture and ansiblejob does not exist, skip this part and creates new ansiblejob
+	// when createNewAnsibleJob is false but ansiblejob does not exist, it creates new ansiblejob
+	if !createNewAnsibleJob {
+		existingAnsibleJob := CheckExistAnsibleJob(policyAutomation, dynamicClient)
+
+		if existingAnsibleJob != nil {
+			log.V(2).Info("Modifiy existing ansiblejob")
+
+			// Update with new policyAutomation's policy_violation
+			existingAnsibleJob.Object["spec"].(map[string]interface{})["extra_vars"] = mapExtraVars
```

Comment:
Creating a new AnsibleJob when more clusters have been violated is the right thing to do because an AnsibleJob might have already triggered a playbook run on Ansible Tower. Updating the `extra_vars` would lead to confusion.

I'm thinking that only your predicate change is needed and not any of the other changes.

Reply:
that means if we have 100 clusters that means we create maybe 90 ansiblejobs  depend on  status changing time it could be 100

Reply:
oh also we need to change previous e2e test too

Reply:
@yiraeChristineKim yes, it's possible. Those AnsibleJob objects do periodically get cleaned up by the Ansible operator though.

The way the controller-runtime reconcile request queue works is that if there are multiple reconcile requests for the same Policy, it consolidates them to 1 request, so in theory, it should be less than 100. For example, if you have a reconcile request for the root policy and while you're creating the AnsibleJob there have been 5 updates to the root policy, you'd only end up with the 1 additional reconcile request for the root policy.

Reply:
oh yeah i guess you right cuz I saw only 2 request when it is 3 cluster changes

Reply:
@yiraeChristineKim let me know when you're ready for another review